src/views/product-landing/helpers/extract-headings.ts

```diff
@@ -0,0 +1,34 @@
+import { TableOfContentsHeading } from 'layouts/sidebar-sidecar/components/table-of-contents'
+import { ProductLandingViewProps } from '../types'
+import { ProductLandingBlock } from '../components/product-landing-blocks/types'
+
+export function extractHeadings(
```

Comment:
Extracts headings from processed content. I'm treating the content processing in `transform-raw-content-to-prop` as the single source of truth for valid headings.